client/src/main/scala/sbtcompendium/client/CompendiumClient.scala

```diff
@@ -102,7 +105,22 @@ object CompendiumClient {
         } yield out
       }
 
-      override def generateClient(target: IdlName, identifier: String): F[String] = F.pure("")
+      override def generateClient(target: IdlName, identifier: String, v: Option[String]): F[List[String]] =
+        target match {
+          case IdlName.Avro =>
+            for {
+              protocol <- retrieveProtocol(identifier, safeInt(v))
+              code     <- handleAvro(protocol)
+            } yield code
+          case _ =>
+            UnknownError(s"Unknown error with status code 501. Schema format not implemented yet")
+              .raiseError[F, List[String]]
+        }
+
+      private def handleAvro(op: Option[Protocol]): F[List[String]] =
+        op.map(p => F.delay(Generator(Standard).stringToStrings(p.raw))).getOrElse(F.pure(Nil))
+
+      private def safeInt(s: Option[String]): Option[Int] = s.flatMap(str => Try(str.toInt).toOption)
```

Comment:
The version could be "1.2.3", right? Or is compendium only supporting integers?

Reply:
Only integers